src/diffusers/image_processor.py

```diff
@@ -537,6 +537,27 @@ def binarize(self, image: PIL.Image.Image) -> PIL.Image.Image:
 
         return image
 
+    def _denormalize_conditionally(
+        self, images: torch.Tensor, do_denormalize: Optional[List[bool]] = None
+    ) -> torch.Tensor:
+        r"""
+        Denormalize a batch of images based on a condition list.
+
+        Args:
+            images (`torch.Tensor`):
+                The input image tensor.
+            do_denormalize (`Optional[List[bool]`, *optional*, defaults to `None`):
+                A list of booleans indicating whether to denormalize each image in the batch. If `None`, will use the
+                value of `do_normalize` in the `VaeImageProcessor` config.
+        """
+        if do_denormalize is None:
+            return self.denormalize(images) if self.config.do_normalize else images
+
+        # De-normalizing a batch and selectively torch.stack'ing the results turns out to be
+        # significantly faster than performing a lot of smaller denormalizations
+        denormalized = self.denormalize(images)
```

Comment:
I think there is some context to that, each image in the batch may have a different value for `do_normalize` for sd1.5, see the code here https://github.com/huggingface/diffusers/blob/6131a93b969f87d171148bd367fd9990d5a49b6b/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L1075

Since most of the new pipelines (sdxl, sd3, flux), we do not pass the `do_normalize` from the pipeline, i.e. `do_normalize is None` here , see SDXL https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L1304 , 

and we already did this so it will be batch processed for all the new pipelines already, I think this line is sufficient here
```
        if do_denormalize is None:
            return self.denormalize(images) if self.config.do_normalize else images
```